### PR TITLE
Update strip_postcommit to remove all possible repl hooks.

### PR DESCRIPTION
**Pull request by jonmeredith**
Previously only the currently enabled hooks were stripped.
If a custom bucket property contained a hook for a disabled
replication mode, then it would be resurrected whenever fixups ran.

Accidentally pushed fix direct to 1.4 branch.  If any issues with reviewing this we will have to fix there.

### CODE REVIEW

File: src/riak_repl.erl [hunk 1]
```diff
@@ -66,5 +69,32 @@ strip_postcommit(BucketProps) ->
...
+
+-define(MY_HOOK1, {struct,
+                    [{<<"mod">>, <<"mymod1">>},
+                     {<<"fun">>, <<"myhook1">>}]}).
+-define(MY_HOOK2, {struct,
+                    [{<<"mod">>, <<"mymod2">>},
+                     {<<"fun">>, <<"myhook2">>}]}).
+strip_postcommit_test_() ->
+    [?_assertEqual(
+        [], % remember, returns the stipped postcommit from bprops
```
Comment (engelsanchez) [inline src/riak_repl.erl:86]: little typo here because I HAD to comment on something

**Comment by cmeiklejohn** [comment]
Should this be tagged for the 2.0-RC milestone?

**Comment by engelsanchez** [comment]
Tagged now. I am verifying and this should be merged pretty soon.

**Comment by cmeiklejohn** [comment]
OK.

**Comment by engelsanchez** [comment]
I've tested this fix and its counterpart already in 1.4 branch (VP of oopsies).  In both cases, you can see how even with everything disabled, removing mode_repl12 from the modes list makes the v2 postcommit hook _reappear_ in the bucket properties, both default and in buckets with custom properties.   With this fix, removing the mode_repl12 mode always ends up removing the v2 hook.

When I say remove, I mean effectively removed by the fixups on local rings. The raw ring still has the hooks, of course. We do not really remove them.

:+1: :boat: :dancer:

**Comment by engelsanchez** [comment]
:+1: 572517e

**Comment by jonmeredith** [comment]
@borshop merge

**Pull request merged**